application/helpers/expressions/em_manager_helper.php

```diff
@@ -8512,6 +8512,8 @@ public static function ProcessCurrentResponses()
                                             }
                                             $aFiles[$i]->filename = $sDestinationFileName;
                                         }
+                                        /* Sanitize size */
+                                        $aFiles[$i]->size = floatval($aFiles[$i]->size);
```

Comment:
@gabrieljenik : maybe we can check the size of the fiilenamle (fu_XXX) at this point and don't use user submitted value ?

Advice ?

(only in master)

Reply:
That would be good too. Always better to have clean data on the DB. Still, not sure if the change is worth to be done now.

Reply:
> Also, another possibility: clean the data when submitted?

I don't understand ? What other data can be cleaned ?

> Still, not sure if the change is worth to be done now.

Maybe a dev/feature mantis

Reply:
> I don't understand ? What other data can be cleaned ?

This size data. But nevermind. I don't thin it worths it.

Reply:
We are in ProcessCurrentResponses : it's **THE** place to clean data before all other process.

The other place it's clean data before save to DB https://github.com/LimeSurvey/LimeSurvey/blob/90197188821c602a5d5dfc2772bbf552f3e7aa01/application/helpers/expressions/em_manager_helper.php#L9739

With checkValidityAnswer : you can add error to the respondant.